cores/lightning-ln882h/arduino/libraries/Serial/Serial.cpp

```diff
@@ -0,0 +1,99 @@
+/* Copyright (c) Etienne Le Cousin 2024-02-10. */
+
+#include "SerialPrivate.h"
+
+extern Serial_t *serial_handles[SER_PORT_NUM];
+
+#if LT_HW_UART0
+SerialClass Serial0(0, PIN_SERIAL0_RX, PIN_SERIAL0_TX);
+
+static void callback_uart0(void) {
+	SerialData *data = (SerialData *)Serial0.data;
+	char ch;
+	while (serial_read(serial_handles[0], &ch, 1)) {
+		data->buf.store_char(ch);
+	}
+}
+#else
+#define callback_uart0 NULL
+#endif
+#if LT_HW_UART1
+SerialClass Serial1(1, PIN_SERIAL1_RX, PIN_SERIAL1_TX);
+
+static void callback_uart1(void) {
+	SerialData *data = (SerialData *)Serial1.data;
+	char ch;
+	while (serial_read(serial_handles[1], &ch, 1)) {
+		data->buf.store_char(ch);
+	}
+}
+#else
+#define callback_uart1 NULL
+#endif
+#if LT_HW_UART2
+SerialClass Serial2(2, PIN_SERIAL2_RX, PIN_SERIAL2_TX);
+
+static void callback_uart2(void) {
+	SerialData *data = (SerialData *)Serial2.data;
+	char ch;
+	while (serial_read(serial_handles[2], &ch, 1)) {
+		data->buf.store_char(ch);
+	}
+}
+#else
+#define callback_uart2 NULL
+#endif
+
+// clang-format off
+const serial_rx_callbcak serial_rx_callbcaks[SER_PORT_NUM] = {
```

Comment:
Typo (yeah, I know it's like this in the SDK...):
```suggestion
const serial_rx_callbcak serial_rx_callbacks[SER_PORT_NUM] = {
```